### PR TITLE
Make it possible to define control buttons text (Back, Create, Review) per template

### PR DESCRIPTION
**Pull request by acierto**
#### :heavy_check_mark: Checklist

There are different types of templates where the different labels for control buttons would make sense. 
But not like a static for all templates. 
In some templates it make sense to have "Create", whenever something new is created. 
But when the flow with help of template is automated to do editing part, "Create" is already confusing. 

Exposing all 3 labels make it more consistent, then just allowing to have only 2 out of 3 (as it is now).

Example of a template with this config: 

```
...
spec:
  owner: backstage/techdocs-core
  type: documentation
  lifecycle: experimental

  configurations:
    buttonLabels:
      backButtonText: 'Previous'
      createButtonText: 'Edit'
      reviewButtonText: 'Verify'
...
```

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
I wonder if we can leverage more of the `i18n` work that's going on here instead. Thinking that you could have one property which points to a subset of translations in there instead of having it in the template.yaml definition instead. 

Thoughts @acierto?

**Comment by acierto** [comment]
@benjdlambert It's not about translations, but more of giving a proper label for a template.

For template 1 is about "Run", another one "Create", third "Edit". Somewhere it make sense to "Delete", if for example template will do some drastic  clean up.

If to leverage here **i18n**, it would still make sense to preserve the granularity per template.
It could look just a bit different like

...
spec:
  owner: backstage/techdocs-core
  type: documentation
  lifecycle: experimental

  configurations:
    buttonLabels:
      backButtonText: ${localized('template.back.label')}
      createButtonText: ${localized('template.back.edit')}
      reviewButtonText: '${localized('template.back.review')}
...

And here I would use still different ${localized('template.back.edit')}, ${localized('template.back.create')}, ${localized('template.back.run')} depends on template's purpose

**Comment by benjdlambert** [comment]
I'm thinking more something like this:

```yaml
spec:
  owner: backstage/techdocs-core
  type: documentation
  lifecycle: experimental
  i18n: deletions
```

Where the i18n file looks like this:

```yaml
deletions:
   createButtonText: 'Delete'
 ```
 
 And the frontend plugin would of course fall back to untranslated stuff for the other options.

**Comment by acierto** [comment]
Ah ok, I see, that make sense.
The only question I have is how would be better to tell to the end user what could be defined there?

**Comment by benjdlambert** [comment]
Yeah, I think that this is an overall an issue for i18n to work out how to surface these options and what is configurable. It's basically just going to be a clone of the default configuration for each individual option in the template.

I think that there was [some work already done for i18n](https://github.com/backstage/backstage/pull/20577) support for the scaffolder, so wondering if it makes sense to build on top of that once that's merged?

**Comment by acierto** [comment]
Ok, i'll wait till i18n work be merged, and build on top of it.

File: plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx
Comment (benjdlambert) [inline plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx:84]: I'm wondering if we should deprecate this way eventually, but can probably leave this for now.

File: plugins/scaffolder-common/api-report.md
Comment (benjdlambert) [inline plugins/scaffolder-common/api-report.md:104]: @Rugvip how's this feeling?

**Review by benjdlambert — COMMENTED** [review]
Thinking this is looking pretty good to be honest - let's just bring in some others to have a quick peek and see if the words are nice :)

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**
